### PR TITLE
Don't retry jobs that are expected to fail

### LINKED ISSUE #4938: [Flaky CI] [sig-network] [Serial]Services bridge interface binding with a service matching the vmi exposed [test_id:1548] should fail to reach the vmi if an invalid servicename is used
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/triage build-watcher
/kind bug

**What happened**:
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/batch/pull-kubevirt-e2e-k8s-1.18/1355038767810875392
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4870/pull-kubevirt-e2e-k8s-1.20/1355087088759869440
```
Job netcat6gl6t timeout reached, status: {Conditions:[] StartTime:2021-01-29 10:41:12 +0000 UTC CompletionTime:<nil> Active:1 Succeeded:0 Failed:0}, err: timed out waiting for the condition
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
At network service testing the timeout for jobs failing or not is 90
seconds and those jobs have three retries, in case the test is expected
to fail 4 differenct containers are going to be create/run/delete since
the job is going to retry it and this reaches the timeout limit (more
often at k8s >= 1.20).

Since on test that expect job to fail retries does not make sense this
change add a parameter to the job creatin function so we specify no
retries if job is expected to fail.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4938

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/network/services.go [hunk 1]
```diff
@@ -103,7 +106,7 @@ var _ = SIGDescribe("[Serial]Services", func() {
 		serviceFQDN := fmt.Sprintf("%s.%s", serviceName, namespace)
 
 		By(fmt.Sprintf("starting a job which tries to reach the vmi via service %s", serviceFQDN))
-		job := runTCPClientExpectingHelloWorldFromServer(serviceFQDN, strconv.Itoa(servicePort), namespace, isIPv6)
+		job := runTCPClientExpectingHelloWorldFromServer(serviceFQDN, strconv.Itoa(servicePort), namespace, isIPv6, 0)
```
Comment (xpivarc) [inline tests/network/services.go:109]: Are we expecting that the connectivity goes away instantly?

Comment (qinqon) [reply tests/network/services.go:109]: If not the job is not going to be marked as Failed

Comment (xpivarc) [reply tests/network/services.go:109]: I see it now.
just a note: It looks like we use it in 2 same tests under different contexts (but the context doesn't change the test case).

**Review by dhiller — APPROVED** [review]
Great job, thanks!

/lgtm
/approve

**Comment by fgimenez** [comment]
Awesome! I think this PR should close https://github.com/kubevirt/kubevirt/issues/4938

**Comment by qinqon** [comment]
/retest 

Not related

File: tests/network/services.go [hunk 2]
```diff
@@ -46,12 +46,15 @@ import (
...
 
-	runTCPClientExpectingHelloWorldFromServer := func(host, port, namespace string, isIPv6 bool) *batchv1.Job {
+	runTCPClientExpectingHelloWorldFromServer := func(host, port, namespace string, isIPv6 bool, retries int32) *batchv1.Job {
 		var pingCmd string
 		if isIPv6 {
 			pingCmd = fmt.Sprintf("ping -c1 %s;", host)
 		}
-		job, err := virtClient.BatchV1().Jobs(namespace).Create(context.Background(), tests.NewHelloWorldJob(host, port, pingCmd), k8smetav1.CreateOptions{})
+		job := tests.NewHelloWorldJob(host, port, pingCmd)
+		job.Spec.BackoffLimit = &retries
```
Comment (maiqueb) [inline tests/network/services.go:55]: My only question here is if this retries parameter should be made available in the original question, but I understand why you only change this private helper. I'm OK with it (I would probably have done the same actually ...).

**Review by maiqueb — APPROVED** [review]
Good catch.

File: tests/network/services.go (hunk 2 above)
Comment (qinqon) [reply tests/network/services.go:55]: I can prepare a follow up with passing retries around in functions, but in locking 1.20 did have more priority.

Comment (maiqueb) [reply tests/network/services.go:55]: Just a thought, I'm ok either way.

**Comment by fgimenez** [comment]
> Awesome! ~I think this PR should close #4938~ not so sure now, need to verify

Checked, with these changes I'm not able to reproduce #4938, without them the error happens after a few executions, more frequent on 1.20.

**Pull request merged**